sideboard/config.py

```diff
@@ -9,14 +10,72 @@
 from validate import Validator
 
 
+_defaults_re = re.compile(r'(.+)-defaults(\.\w+)$')
+
+
+def uniquify(x):
+    """
+    Returns an ordered copy of `x` with duplicate items removed.
+
+    >>> uniquify(['a', 'b', 'a', 'c', 'a', 'd', 'a', 'e'])
+    ['a', 'b', 'c', 'd', 'e']
+
+    """
+    seen = set()
+    return [i for i in x if i not in seen and not seen.add(i)]
```

Comment:
> It really belongs in sideboard.lib._util, but trying to import it from there causes a circular import error (even if I import it from inside a function).

That makes sense because ``sideboard.lib`` assumes that ``sideboard.config`` has already been implemented.  However, if we decide we want it to be importable in ``sideboard.lib`` then we could import it in ``sideboard/lib/__init__.py`` and then add it to that module's ``__all__``.  However, I'm not sure this is generally useful enough for us to put it there.

> In my original implementation I DID call listify on the argument! But then I thought, if someone passes in an argument that is not already "listy" we should probably error on that. For example, if someone accidentally passes in None, then we would return [None], which I don't think is what we intend to do. So, for my perspective, if that's really what someone wants, then they can listify the argument before passing it to uniquify.

Reasonable!  In that case I suggest importing ``sideboard.lib.is_listy`` adding a line like

```python
assert is_listy(xs), 'uniquify requires a listy argument'
```

> I used x as the parameter name because that's what we also use for listify and is_listy, but then I was like, "what variable am I gonna use for the items? Oh well, i it is!"

Also reasonable.  The reason why we use ``x`` for ``listify`` and ``is_listy`` is because we're not sure whether it's already an iterable or not, so I thought ``x`` made more sense than ``xs`` there.